R/module_tabs_with_filters.R

```diff
@@ -99,6 +99,11 @@
 ui_tabs_with_filters <- function(id, modules, datasets, filter) {
   checkmate::assert_class(modules, "teal_modules")
   checkmate::assert_list(datasets, types = c("list", "FilteredData"))
+  if (!missing(filter)) {
+    checkmate::assert_class(filter, "teal_slices")
+  } else {
+    filter <- teal_slices(module_specific = FALSE)
```

Comment:
what about specifying default value for filter as `teal_slices(module_specific = FALSE)`? no strong opinion on it.
srv_tabs_with_filters has filter se to default value and this could be set to `teal_slices(module_specific = FALSE)` instead of NULL

Reply:
This function has no defaults at all.

Reply:
wanna remove default value for filter in srv_tabs_with_filters for consistency then?

Reply:
No, those seem to be there for a reason. I could consider having `filter` like in the UI function but then the example has to be changed as well, which I don't want to do out of respect for the author. The way `reporter` is passed down the stack is shady and that default may be crucial.

Reply:
I hear you